data_crawler/crawler.py

```diff
@@ -75,14 +64,32 @@ def get_session(
             List of status codes that will trigger a retry.
         """
         retries = urllib3.Retry(
-            total=total_retries, 
-            backoff_factor=backoff_factor, 
-            status_forcelist=status_forcelist
+            total=self.total_retries, 
+            backoff_factor=self.backoff_factor, 
+            status_forcelist=self.status_forcelist
         )
         adapter = HTTPAdapter(max_retries=retries)
-        self.session = requests.Session()
-        self.session.mount('http://', adapter)
+        session = requests.Session()
+        session.mount('http://', adapter)
 
+        return session
+        
+    def get_url(self, url: str): 
+        """Get the response from a URL.
+
+        Parameters
+        ----------
+        url : str
+            The URL to get the response from.
+        """
+        
+        response = self.session.get(url)
+        try:
+            response.raise_for_status()
+        except (HTTPError, RequestException, ConnectionError) as err:
+            self.logger.error(f'Error when connecting: {err}')
+        else:
+            return response
 
     def translate_text(self, text):
```

Comment:
For this Translator function, I would like to propose changes such as inputting the original language of the text supplied from the main flow. This could be seen on my universityadmission crawler in the archive folder. This could help to reduce the number of interactions with GGTranslator and if the lang == en --> just return it as is.

Reply:
agree